frontend/src/components/editor/chrome/panels/datasources-panel.tsx

```diff
@@ -0,0 +1,391 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+import React from "react";
+import {
+  ChevronRightIcon,
+  DatabaseIcon,
+  PlusSquareIcon,
+  XIcon,
+} from "lucide-react";
+import type { DataTable, DataTableColumn } from "@/core/network/types";
+import { Command, CommandInput, CommandItem } from "@/components/ui/command";
+import { CommandList } from "cmdk";
+
+import { cn } from "@/utils/cn";
+import { useDatasets, useDatasetsActions } from "@/core/datasets/state";
+import { DATA_TYPE_ICON } from "@/components/datasets/icons";
+import { Button } from "@/components/ui/button";
+import { useCellActions } from "@/core/cells/cells";
+import { lastFocusedCellIdAtom } from "@/core/cells/focus";
+import { useAtomValue } from "jotai";
+import { Tooltip } from "@/components/ui/tooltip";
+import { PanelEmptyState } from "./empty-state";
+import { previewDatasetColumn } from "@/core/network/requests";
+import type { ColumnPreviewMap } from "@/core/datasets/types";
+import { prettyNumber } from "@/utils/numbers";
+import { Events } from "@/utils/events";
+import { CopyClipboardIcon } from "@/components/icons/copy-icon";
+import { ErrorBoundary } from "../../boundary/ErrorBoundary";
+import type { TopLevelFacetedUnitSpec } from "@/plugins/impl/data-explorer/queries/types";
+import { useTheme } from "@/theme/useTheme";
+import {
+  maybeAddAltairImport,
+  maybeAddMarimoImport,
+} from "@/core/cells/add-missing-import";
+import { autoInstantiateAtom } from "@/core/config/config";
+import { DataColumnPreview } from "@/core/kernel/messages";
+
+export const DataSourcesPanel: React.FC = () => {
```

Comment:
Could it be possible to suggest or even prompt users to install `altair` if it's not installed, to get plots?

Doesn't have to be in this PR, but could be a good follow-up to make the plots feature more discoverable.

Reply:
easy to add with how we send back errors - i can do in a followup though